cmd/spiffe-aws-assume-role/cli/cli_context.go

```diff
@@ -12,3 +12,24 @@ type CliContext struct {
 	Logger            *logrus.Logger
 	Telemetry         *telemetry.Telemetry
 }
+
+func NewDefaultCliContext() (*CliContext, error) {
+	logger := logrus.New()
+	// Example log line:
+	// time="2021-03-01T16:28:51-08:00" level=error msg="failed to parse SPIFFE ID from 305d07ed-765e-4642-9bd3-4c74aa86f5ed: spiffeid: invalid scheme"
+	logger.SetFormatter(&logrus.TextFormatter{})
+
+	telemetry, err := telemetry.NullTelemetry()
+	if err != nil {
+		return nil, err
+	}
+
+	context := &CliContext{
+		JWTSourceProvider: credentials.StandardJWTSourceProvider,
+		STSProvider:       credentials.StandardSTSProvider,
+		Logger:            logger,
+		Telemetry:         telemetry,
+	}
+
+	return context, nil
+}
```

Comment:
You can think of the execution of the CLI as having 4 steps:

1. Create CliContext
2. Parse command line arguments into Kong context
3. Invoke Run on Kong context, passing in CliContext
4. Actual business logic

We'd like to allow callers to perform steps 1-3 more easily from their own code, giving them the opportunity to inspect the results of parsing the command line (and possibly modify them) before beginning the actual business logic. This change exposes the necessary machinery to make this possible.

Reply:
So users who want to modify the CliContext would define their own version of `RunWithDefaultContext` and call that in main.go?

Reply:
Yeah, that's more or less exactly what I'd prototyped. Another option would be to add a callback but then we've got yet another overload of Run, and I think we've already got plenty of those.